### PR TITLE
WIP: embed base64 encoded event data in the pty itself

### PR DESCRIPTION
**Pull request by ryanpetrello**
this removes the `-partial.json` IPC mechanism for passing event data
from ansible to the runner process

### CODE REVIEW

**Comment by ryanpetrello** [comment]
This doesn't pass tests yet, but I think it should work.

File: ansible_runner/utils.py [hunk 1]
```diff
@@ -326,6 +306,19 @@ def _emit_event(self, buffered_stdout, next_event_data=None):
             event_data['end_line'] = self._start_line + n_lines
             self._start_line += n_lines
             if self._event_callback:
+                if event_data.get('stdout'):
```
Comment (matburt) [inline ansible_runner/utils.py]: Even *if* this PR doesn't work out for replacing partials we should probably still do this. Trying to emit stdout down at the callback plugin gives us all kinds of grief.

File: ansible_runner/runner.py [hunk 2]
```diff
@@ -22,8 +22,8 @@
 
 class Runner(object):
 
-    def __init__(self, config, cancel_callback=None, remove_partials=True,
-                 event_handler=None, finished_callback=None, status_handler=None):
+    def __init__(self, config, cancel_callback=None, event_handler=None,
```
Comment (matburt) [inline ansible_runner/runner.py]: Partial deletion control is also a parameterization of the module that we'll need to remove also.

**Comment by ryanpetrello** [comment]
I tested this playbook:

```yaml
- name: run setup
  hosts: all
  tasks:
    - setup:
        gather_subset: all
```

...with 10 hosts and `--forks=10`, and the stdout and number of generated job events matched before _and_ after this PR.

The only concern I have with this PR is one that @matburt expressed - if there's any potential for parallel processes to interleave this stdout.

File: ansible_runner/runner.py (hunk 2 above)
Comment (ryanpetrello) [reply ansible_runner/runner.py]: Where?

File: ansible_runner/utils.py (hunk 1 above)
Comment (ryanpetrello) [reply ansible_runner/utils.py]: Agreed, this is a much saner place to handle `sys.stdout`

File: ansible_runner/runner.py (hunk 2 above)
Comment (matburt) [reply ansible_runner/runner.py]: hmm, I guess I'm wrong about this one.

**Comment by ryanpetrello** [comment]
Haven't found time for this, might reinvestigate later.

**Pull request closed without merging**